### PR TITLE
Add generated riff cli docs to README

### PR DESCRIPTION
**Pull request by scothis**
- converted readme to markdown
- generate docs for `riff init java` and `riff create java` in
  docs directory
- embed command docs into the README
- fail CI if the generated docs drift from what's committed

Refs: projectriff/riff#454

### CODE REVIEW

**Comment by dsyer** [comment]
Does this make #52 redundant? You could have said. Humph.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+Initialize a java function
+
+#### Synopsis
+
+Generate the function based on the function source code specified as the filename, using the artifact (jar file),
+the function handler(classname), the name and version specified for the function image repository and tag.
+
+For example, from a maven project directory named 'greeter', type:
+
+    riff init java java -i greetings -a target/greeter-1.0.0.jar --handler=Greeter
```
Comment (trisberg) [inline README.md]: java java?

**Comment by scothis** [comment]
@dsyer nope, the conversion of the readme to markdown is the least interesting part of this PR. Happy to rebase after you merge #52.

File: README.md (hunk 1 above)
Comment (scothis) [reply README.md]: that's a bug in the current java-invoker.yaml, will fix

**Comment by trisberg** [comment]
Is #52 ready for merge?

**Pull request merged**